main.tf

```diff
@@ -1,46 +1,68 @@
 locals {
-  datacenters     = join(",", var.nomad_datacenters)
-  buckets         = var.hive_bucket # output variable for presto
-  hive_env_vars   = join("\n",
+  datacenters   = join(",", var.nomad_datacenters)
+  buckets       = var.hive_bucket # output variable for presto
+  hive_env_vars = join("\n",
     concat([
       "JUST_EXAMPLE_ENV=some-value",
     ], var.hive_container_environment_variables)
   )
+  vault_provider = var.postgres_vault_secret.use_vault_provider || var.minio_vault_secret.use_vault_provider
```

Comment:
Maybe a little obvious, but I want to be sure. You are basing whether or not you should use vault on whether or not postgres or minio uses it, correct? Just resolve this is yes!